include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -176,10 +185,67 @@ def RepeatOp : LTLOp<"repeat", [Pure]> {
   }];
 }
 
+def GoToRepeatOp : LTLOp<"goto_repeat", [Pure]> {
+  let arguments = (ins
+    LTLAnySequenceType:$input,
+    I64Attr:$base,
+    I64Attr:$more);
+  let results = (outs LTLSequenceType:$result);
+  let assemblyFormat = [{
+    $input `,` $base (`,` $more^)? attr-dict `:` type($input)
+  }];
+  let hasFolder = 1;
+
+  let summary = "`goto`-style non-consecutively repeating sequence.";
+  let description = [{
+    Non-consecutive repetition of the `$input` sequence. This must hold between `$base` 
+    and `$base + $more` times in a finite number of evaluations. The final evaluation 
+    in the sequence has to match. The `$base` must be greater than or equal to zero 
+    and the range `$more` can't be omitted. For example, a !b b b !b !b b c represents 
+    a matching observation of `ltl.goto_repeat %b, 1, 2`, but a !b b b !b !b b !b c doesn't.
+  }];
+}
+
+def NonConsecutiveRepeatOp : LTLOp<"non_consecutive_repeat", [Pure]> {
+  let arguments = (ins
+    LTLAnySequenceType:$input,
+    I64Attr:$base,
+    I64Attr:$more);
+  let results = (outs LTLSequenceType:$result);
+  let assemblyFormat = [{
+    $input `,` $base (`,` $more^)? attr-dict `:` type($input)
+  }];
+  let hasFolder = 1;
+
+  let summary = "`goto`-style non-consecutively repeating sequence.";
+  let description = [{
+    Non-consecutive repetition of the `$input` sequence. This must hold between `$base` 
+    and `$base + $more` times in a finite number of evaluations. The final evaluation 
+    in the sequence does not have to match. The `$base` must be greater than or equal to zero 
+    and the range `$more` can't be omitted. For example, both a !b b b !b !b b c and 
+    a !b b b !b !b b !b c represent matching observations of 
+    `ltl.non_consecutive_repeat %b, 1, 2`.
+  }];
+}
+
+
 //===----------------------------------------------------------------------===//
 // Properties
 //===----------------------------------------------------------------------===//
 
+def MatchOp : LTLOp<"match", [Pure]> {
+  let arguments = (ins LTLAnyPropertyType:$input);
+  let results = (outs I1:$result);
+  let assemblyFormat = [{
+    $input attr-dict `:` type($input)
+  }];
+  let summary = "Checks if the given property mathces in the current cycle.";
+  let description = [{
+    Given a property `$input`, this operation evaluates it in the current cycle 
+    and returns and i1 that is the result of that evaluation.
+  }];
+}
```

Comment:
This is a cool op to have, because it bridges the world of properties and booleans. How would this lower to SVA?

Reply:
I was about to write something that made sense to me, but now that I think about this would basically lead to us needed a synthesised clone of the property. Not sure anymore, I will need to think about this a bit more.